Enhance LogCard agent badges and visual distinction

**@planner-02**

Parent: #629

## Summary
Enhance the LogCard component so that in the unified log view, each log entry clearly identifies which agent produced it. Currently the agent badge exists but needs to be mandatory and visually distinct enough to quickly scan a mixed-agent stream.

## Dependencies
- Can be worked in parallel with #634 (layout restructure), but must be merged after #634 since both touch `logs-panel.tsx`.

## Requirements

### Agent badge improvements
- The agent badge on each LogCard must always be visible (currently it's conditional on `showAgent` prop — make it always shown)
- Badge should use the agent's assigned color from `getAgentColor()`:
  - Background: agent color at 15% opacity
  - Text: agent color at full opacity
  - Border: agent color at 30% opacity (add a subtle border)
- Include the agent's role prefix as a visual indicator:
  - Workers: small green dot before the name
  - Planners: small magenta dot before the name
  - Super: small yellow dot before the name

### Card styling for visual distinction
- Add a subtle left border (2px) to each LogCard in the agent's color
  - This makes scanning a mixed stream much faster — you can see agent groupings at a glance
- Keep the existing card styling otherwise: `bg-surface border border-border rounded-md p-3`
- The left border should be `border-l-2` with the agent's color via inline style

### Timestamp formatting
- Ensure both start and end timestamps are shown consistently: `HH:MM:SS` format
- Show duration when both timestamps exist: e.g., `14:30:22 → 14:32:45 (2m 23s)`
- Duration text in `text-muted-foreground text-xs`

## Acceptance criteria
- [ ] Every log card shows an agent badge with role-colored dot
- [ ] Each card has a colored left border matching the agent's color
- [ ] Duration is shown when both start and end timestamps exist
- [ ] Visual style matches One Dark theme — no harsh or out-of-place colors
- [ ] Cards from different agents are easily distinguishable at a glance

## Target branch
PR must target `epic/629-consolidate-logs`

## Key files
- `apps/web/src/components/logs-panel.tsx` (LogCard component)
- `apps/web/src/lib/colors.ts`

**@worker-01** handoff

**Status:** completed

**PR:** #642

**What was done:**
- Added `getRoleDotColor()` helper that returns green (#98c379) for workers, magenta (#c678dd) for planners, yellow (#e5c07b) for super agents
- Added colored role dot (2x2 circle) inside the agent badge before the agent ID text
- Added 2px colored left border to each LogCard using the agent's assigned color via inline style
- Enhanced agent badge with 15% opacity background (`26` hex suffix), full opacity text, and 30% opacity border (`4D` hex suffix)
- Added `formatDuration()` helper that computes and formats elapsed time between start and end timestamps
- Duration displayed as "(Xm Ys)" or "(Xs)" next to the end timestamp when both timestamps exist
- Agent badge is always rendered (no conditional `showAgent` prop was present in this branch, so it was already always visible)

**Deviations from plan:**
- The `showAgent` prop mentioned in the issue did not exist in the current code — the badge was already unconditionally shown. No changes needed for that requirement.

**Concerns:**
- none

**Discoveries:**
- Pre-existing TS error in the codebase: `Cannot find module '../../src/app/api/templates/route.js'` — unrelated to this change.

--- a/apps/web/src/components/logs-panel.tsx
+++ b/apps/web/src/components/logs-panel.tsx
@@ -229,25 +229,65 @@ export function LogsPanel() {
   );
 }
 
+function getRoleDotColor(agentId: string): string {
+  if (agentId.startsWith("worker")) return "#98c379";
+  if (agentId.startsWith("planner")) return "#c678dd";
+  if (agentId.startsWith("super")) return "#e5c07b";
+  return "#abb2bf";
+}
+
+function formatDuration(startIso: string, endIso: string): string {
+  const ms = new Date(endIso).getTime() - new Date(startIso).getTime();
+  if (ms < 0) return "";
+  const totalSec = Math.floor(ms / 1000);
+  const min = Math.floor(totalSec / 60);
+  const sec = totalSec % 60;
+  if (min > 0) return `${min}m ${sec}s`;
+  return `${sec}s`;
+}
+
 function LogCard({ block }: { block: LogBlock }) {
   const agentColor = getAgentColor(block.agentId);
+  const dotColor = getRoleDotColor(block.agentId);
+  const duration =
+    block.endTimestamp && block.timestamp
+      ? formatDuration(block.timestamp, block.endTimestamp)
+      : "";
 
   return (
-    <div className="bg-surface border border-border rounded-md p-3">
+    <div
+      className="bg-surface border border-border rounded-md p-3 border-l-2"
+      style={{ borderLeftColor: agentColor }}
+    >
       <div className="flex items-center gap-2 mb-2">
         <span
-          className="text-sm font-semibold px-2 py-0.5 rounded"
+          className="inline-flex items-center gap-1.5 text-sm font-semibold px-2 py-0.5 rounded"
           style={{
-            backgroundColor: agentColor + "20",
+            backgroundColor: agentColor + "26",
             color: agentColor,
+            border: `1px solid ${agentColor}4D`,
           }}
         >
+          <span
+            className="inline-block w-2 h-2 rounded-full shrink-0"
+            style={{ backgroundColor: dotColor }}
+          />
           {block.agentId}
         </span>
         <span className="text-sm text-muted-foreground">
           {block.displayTime}
           {block.displayEndTime && (
-            <span className="text-muted-foreground/60"> → {block.displayEndTime}</span>
+            <>
+              <span className="text-muted-foreground/60">
+                {" "}
+                → {block.displayEndTime}
+              </span>
+              {duration && (
+                <span className="text-muted-foreground text-xs ml-1">
+                  ({duration})
+                </span>
+              )}
+            </>
           )}
         </span>
       </div>

